pkg/sleep/sleep.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sleep
+
+import (
+	"context"
+	"time"
+)
+
+// Context sleeps for the given duration, or until the context is complete.
+// Returns true if the sleep completes the full duration
+func Context(ctx context.Context, d time.Duration) bool {
```

Comment:
This name is confusing, indicates it returns a context.

Reply:
Any suggestions? When you call it's `sleep.Context()` so it was pretty clear to me.

Reply:
How about removing this function. ctx.Done() is simple to call

Reply:
Either remove or name `UntilContext` ?

Reply:
UntilContext may be better?